src/main/java/com/targus/ui/controllers/AlgorithmSelectionController.java

```diff
@@ -0,0 +1,58 @@
+package com.targus.ui.controllers;
+
+import com.targus.utils.Constants;
+import javafx.animation.KeyFrame;
+import javafx.animation.KeyValue;
+import javafx.animation.Timeline;
+import javafx.fxml.FXML;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.util.Duration;
+
+public class AlgorithmSelectionController {
+    public ComboBox algorithmComboBox;
+    public VBox GAInputs;
+    public StackPane algorithmInputs;
+    public ComboBox mutationComboBox;
+    public TextField mutationRateTextField;
+    public ComboBox terminationComboBox;
+    public TextField terminationTextField;
+
+    public String getAlgorithm() {
+        return algorithmComboBox.getValue().toString();
+    }
+
+    public String getMutation() {
+        return mutationComboBox.getValue().toString();
+    }
```

Comment:
I think we should have a check here for different algorithms. For instance, this only makes sense for StandardGA and ImprovedGA. I am aware that we do not call this method for other algorithms but adding one if statement does not hurt. I am suggesting this:

```suggestion
    public String getMutation() {
        if (algorithmComboBox.getValue().toString() != "StandardGA" || algorithmComboBox.getValue().toString() != "ImprovedGA") {
            return "";
        }
        return mutationComboBox.getValue().toString();
    }
```